File: TENavigaaBackIcon/TENavigaaBackIcon/ViewController.m
//
//  ViewController.m
//  TENavigaaBackIcon
//
//  Created by bo on 18/09/2017.
//  Copyright © 2017 SAM. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    
    UIView *redv = [UIView new];
    redv.backgroundColor = [UIColor redColor];
    
    UIView *blurv = [UIView new];
    blurv.backgroundColor = [UIColor blueColor];
    
    UIView *yelv = [UIView new];
    yelv.backgroundColor = [UIColor yellowColor];
    
    [self.view addSubview:redv];
    [self.view addSubview:blurv];
    [self.view addSubview:yelv];
    
    [redv setTranslatesAutoresizingMaskIntoConstraints:NO];
    
    NSLayoutConstraint *c1 = [redv.topAnchor constraintEqualToAnchor:self.view.readableContentGuide.bottomAnchor];
    NSLayoutConstraint *c2 = [redv.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor];
    NSLayoutConstraint *c3 = [redv.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor];
    NSLayoutConstraint *c4 = [redv.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor];
    
    [NSLayoutConstraint activateConstraints:@[c1,c2,c3,c4]];
    
    self.navigationItem.backBarButtonItem = [[UIBarButtonItem alloc] initWithImage:nil style:UIBarButtonItemStylePlain target:nil action:nil];
    
//    [self.view addConstraints:@[c1,c2,c3,c4]];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: TESwiftXcode9Test/TESwiftXcode9Test/ViewController.swift
//
//  ViewController.swift
//  TESwiftXcode9Test
//
//  Created by bo on 06/06/2017.
//  Copyright © 2017 TE. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        
        let numberDictionary = ["one": 1, "two": 2, "three": 3, "four": 4]
        let evenOnly = numberDictionary.lazy.filter { (_, value) in
            value % 2 == 0
        }
        
        var viaIteration: [String: Int] = [:]
        for (key, value) in evenOnly {
            viaIteration[key] = value
        }
        
        let viaReduce: [String: Int] = evenOnly.reduce([:]) { (cumulative, kv) in
            var dict = cumulative
            dict[kv.key] = kv.value
            return dict
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

